### PR TITLE
JDK-8292066 Convert TestInputArgument.sh and TestSystemLoadAvg.sh to java version

### PR DESCRIPTION
**Pull request by bwhuang-us**
This task converts 2 shell tests to java version. 
test/java/lang/management/OperatingSystemMXBean/TestSystemLoadAvg.sh 
test/java/lang/management/RuntimeMXBean/TestInputArgument.sh

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8292066](https://bugs.openjdk.org/browse/JDK-8292066): Convert TestInputArgument.sh and TestSystemLoadAvg.sh to java version

### Reviewers
 * [Leonid Mesnik](https://openjdk.org/census#lmesnik) (@lmesnik - **Reviewer**)
 * [Alex Menkov](https://openjdk.org/census#amenkov) (@alexmenkov - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/9848/head:pull/9848` \
`$ git checkout pull/9848`

Update a local copy of the PR: \
`$ git checkout pull/9848` \
`$ git pull https://git.openjdk.org/jdk pull/9848/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 9848`

View PR using the GUI difftool: \
`$ git pr show -t 9848`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/9848.diff">https://git.openjdk.org/jdk/pull/9848.diff</a>

</details>

### CODE REVIEW

**Comment by bwhuang-us** [comment]
@lmesnik Could you please review this PR?

File: test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java [hunk 1]
```diff
@@ -54,25 +58,25 @@ public class GetSystemLoadAverage {
...
+    @BeforeMethod
+    void beforeMethod() throws InterruptedException {
+        System.out.println("Wait for 5 seconds.");
+        Thread.sleep(5000);
+    }
 
-        if (args.length == 0) {
+    @Test(invocationCount = 5, timeOut = 300)
+    void testSystemLoadAvg() throws Exception {
+        if (!OS.contains("Win")) {
```
Comment (lmesnik) [inline test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java:71]: Check /test/lib/jdk/test/lib/Platform.java, it contains "Platform.isWindows()" which could be used for this.

File: test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java [hunk 2]
```diff
@@ -54,25 +58,25 @@ public class GetSystemLoadAverage {
     // Allow some delta.
     private static double DELTA = 0.05;
```
Comment (lmesnik) [inline test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java:59]: Not a part of your fix but could you make it final?

File: test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java [hunk 3]
```diff
@@ -54,25 +58,25 @@ public class GetSystemLoadAverage {
...
+    private static final String OS = System.getProperty("os.name");
+
+    @BeforeMethod
+    void beforeMethod() throws InterruptedException {
+        System.out.println("Wait for 5 seconds.");
+        Thread.sleep(5000);
+    }
 
-        if (args.length == 0) {
+    @Test(invocationCount = 5, timeOut = 300)
```
Comment (lmesnik) [inline test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java:69]: I am not sure it is the correct replacement. Accordingly to TestNG doc the invocationCount = 5 means that TestNG calls the test 5 times.  And test fails if any of the invocations fail while the bash script makes 5 attempts and passes if testcase passed in any of them.

Comment (bwhuang-us) [reply test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java:69]: Good catch! I've missed the exit statement in the original shell script.

File: test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java [hunk 4]
```diff
@@ -52,31 +58,48 @@ public class GetSystemLoadAverage {
...
+    private static final int WAIT_TIME_MS = 5000;
+
+    @Test(timeOut = (300 + WAIT_TIME_MS) * MAX_RETRIES)
+    void testSystemLoadAvg() throws Exception {
+        for (int i = 1; i <= MAX_RETRIES; i++) {
+            try {
+                System.out.println(
+                        String.format("Run %d: TestSystemLoadAvg", i));
+                if (!Platform.isWindows()) {
+                    // On Linux or Solaris
```
Comment (lmesnik) [inline test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java:73]: Sorry, if missed the first time. It is Linux or Mac, or better something like *mix.

File: test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java [hunk 5]
```diff
@@ -52,31 +58,48 @@ public class GetSystemLoadAverage {
...
+                        String.format("Run %d: TestSystemLoadAvg test passed",
+                                i));
+                return;
+            } catch (Exception e) {
+                System.out.println(
+                        String.format("TEST FAILED: TestSystemLoadAvg test " +
+                                        "failed %d runs",
+                                i));
+                if (i == MAX_RETRIES)
+                {
```
Comment (lmesnik) [inline test/jdk/java/lang/management/OperatingSystemMXBean/GetSystemLoadAverage.java:95]: please move move brace to the prev line

**Review by lmesnik — APPROVED** [review]

**Review by alexmenkov — APPROVED** [review]

**Pull request closed without merging**